rust-codegen/codegen/src/types.rs

```diff
@@ -0,0 +1,1706 @@
+use std::io::{BufReader, BufWriter, Read, Write};
+use std::path::{Path, PathBuf};
+use std::fs::File;
+use std::collections::HashSet;
+
+use errors::{Error, Result};
+use parser::file_descriptor;
+use keywords::sanitize_keyword;
+
+use std::str;
+
+use heck::SnakeCase;
+// use heck::CamelCase;
+
+
+// TODO: could we code gen reference handling based on the Blob convention
+// all blobs could be content hased by the library + stored as such
+// they could be validated and re-encoded also by the library
+
+pub fn snake_case(s: &String) -> String {
```

Comment:
There's a lot of dead code in this file. We talked in person about making sure this code is warning-free, which should end up meaning this dead code is removed.